src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/ControlPlaneDataStore.cs

```diff
@@ -153,5 +270,52 @@ private async Task<T> UpsertSystemObjectAsync<T>(T systemObject, string partitio
                 throw;
             }
         }
+
+        private async Task<string> DeleteSystemDocumentByIdAsync<T>(string id, string partitionKey, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(id, nameof(id));
+            EnsureArg.IsNotNull(partitionKey, nameof(partitionKey));
+            var eTagAccessCondition = new AccessCondition();
+
+            try
+            {
+                var documentUri = UriFactory.CreateDocumentUri(_databaseId, _collectionId, id);
+
+                var requestOptions = new RequestOptions
+                {
+                    PartitionKey = new PartitionKey(partitionKey),
+                    AccessCondition = eTagAccessCondition,
+                };
+
+                await _documentClient.Value.DeleteDocumentAsync(documentUri.ToString(), requestOptions, cancellationToken);
+
+                return "success";
+            }
+            catch (DocumentClientException dce)
+            {
+                if (dce.StatusCode == HttpStatusCode.TooManyRequests)
+                {
+                    throw new RequestRateExceededException(dce.RetryAfter);
+                }
+
+                _logger.LogError(dce, "Unhandled Document Client Exception");
+                throw;
+            }
+        }
+
+        private void ValidateRole(Role role)
+        {
+            var issues = new List<OperationOutcome.IssueComponent>();
```

Comment:
Operation outcome is a fhir specific thing and shouldn't be used here. There's a little bit of refactoring in #315 that will make this not needed.